### ISSUE TITLE
Clarify upgrade process

### ISSUE THREAD

**Issue by nqb** [YunoHost-Apps/wordpress_ynh#124]
Hello,

TL;DR: we need to keep upgrade process before #119 with a better explanation to users on how to proceed.

I'm using 5.6~ynh1. I recently upgraded to this version and noticed behavior [described on the forum](https://forum.yunohost.org/t/wordpress-beautiful-website-blog-or-app/3961/58) by @maniackcrudelis: 

If you upgrade your YunoHost Wordpress app from 5.4 to 5.6, you need to manually upgrade WordPress to 5.6 using WordPress web admin. Otherwise, minor upgrades are covered by Companion Update plugin installed with WordPress by YunoHost packaging.

This behavior was not really documented here except by a comment in README.md (Limitations section) (see #58).

I'm using YunoHost app since 1 year and I didn't notice any issue with Companion Update plugin so I'm not sure [assessment on instability of plugin](https://github.com/YunoHost-Apps/wordpress_ynh/pull/58#issuecomment-473660107) is still true. Perhaps @Thatoo and @maniackcrudelis have an opinion on this topic.

On the other side, I noticed changes introduced by #119. If I understand correctly, it means that now new installations and current installation that upgrade to 5.6~ynh2 will have automatic upgrades of WordPress core for major versions.

At first glance, I find this config too agressive. It means that if WordPress release a 5.7, 5.6~ynh2 installation will automatically upgrade to 5.7. I see several drawbacks:
- You have a YunoHost package version that didn't match your WordPress version
- If 5.7 comes with specific dependencies, there will be not installed by automatic upgrade
- A major release could change several things that need to be handle by packaging

So I suggest we keep previous behavior:
- minor upgrades are done automatically by Companion Update plugin
- major upgrades need to be done in two steps:
  1. Yunohost package
  1. WordPress web admin

and add notes in README.md

**Comment by kay0u** [comment]
Hi!

Thank you to rise this question.

First of all, I don't use WordPress and I took the maintainer role because I know it's a popular application and I didn't want it not to be maintained.

I ~recently discovered that an update of WordPress in YunoHost didn't really trigger an update, and that it had to be done by hand in the WordPress admin panel. I think that for new users, it can be confusing to have an update that doesn't really update the app

> If I understand correctly, it means that now new installations and current installation that upgrade to 5.6~ynh2 will have automatic upgrades of WordPress core for major versions.

Yes that's my idea since the current behavior is considered as a bug of the Companion Update plugin (If I understand well this [comment](https://github.com/YunoHost-Apps/wordpress_ynh/pull/58#issuecomment-473660107)).

>     * You have a YunoHost package version that didn't match your WordPress version

This is currently the case when we *update* WordPress, YunoHost displays the wrong version if the admin has not manually triggered the update.

>     * If 5.7 comes with specific dependencies, there will be not installed by automatic upgrade

It's true! But since WordPress is used by a lot of people, is this really happening?

>     * A major release could change several things that need to be handle by packaging

For now, in the upgrade script I can't see anything like that.

To be honest when I wrote #119, I was thinking of this case, and I figured that if users knew what they were doing, they could disable the automatic update of major versions. (that's why I make it only for package updates < 5.6~ynh2, and not for each update). But I admit it, it probably lacks a warning during this update and also some information in the Readme.

But if you are an experienced Wordpress user, you probably know what is best to do about these updates, and I'm not against the idea to keep it as it was before my PR #119, with more warnings when there is a major update to be done.
